### ISSUE TITLE
Fixing by `prefer-object-spread` breaks indentation

### ISSUE THREAD

**Issue by berlysia** [palantir/tslint#3025]
### Bug Report

- __TSLint version__: 5.5.0
- __TypeScript version__: 2.1.6
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
const obj1 = {a: 1};
const obj2 = {b: 2};

const merged = Object.assign(
  {},
  obj1,
  {
    a: 11,
    b: 22,
  },
  obj2
);
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "prefer-object-spread": true
  }
}
```

#### Actual behavior
After fixing, I got:

```ts
const merged = {...obj1,
  
    a: 11,
    b: 22,
  ...obj2
};
```

#### Expected behavior
```ts
const merged = {
  ...obj1, 
  a: 11,
  b: 22,
  ...obj2
};
```

**Comment by ajafff** [comment]
I know the current behavior is a PITA.
Making the fixer output of rules compatible with any other rule you may have enabled is not a very high priority.
In fact it would be really hard to get the indentation right, especially for the nested properties `a` and `b` in your example.

**Comment by berlysia** [comment]
> In fact it would be really hard to get the indentation right, especially for the nested properties `a` and `b` in your example.

I have the same opinion. It seems to be hard.
However, the current output is unshapely. I think this rule might be better to drop fixability.

**Comment by ajafff** [comment]
> However, the current output is unshapely. I think this rule might be better to drop fixability.

It works nice for single line expressions. And besides the code style issues, it works in all scenarios. I don't think that this justifies removing the fixer.

When I find the time, I'll look into this and fix the output to something like this:

```ts
const merged = {
  ...obj1,
    a: 11,
    b: 22,
  ...obj2
};
```
The align issue will be fixed by the `align` rule afterwards (or on a second run).

**Comment by berlysia** [comment]
That's a good idea. Simple but worthy.

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
